pip_outdated/check_outdated.py

```diff
@@ -9,60 +9,79 @@
 
 from .verbose import verbose
 
+class Release(TypedDict):
+    version: Version
+    date: str
 
 @dataclass
 class OutdateResult:
     requirement: Requirement
     version: Optional[Version]
-    all_versions: List[Version]
+    all_releases: List[Release]
 
-    wanted: Optional[Version] = None
-    latest: Optional[Version] = None
+    wanted_version: Optional[Version] = None
+    wanted_date: Optional[str] = None
```

Comment:
Instead of making a new variable, we should just change the type i.e.
```py
wanted: Optional[Version]
latest: Optional[Version]
```
to
```py
wanted: Optional[Release]
latest: Optional[Release]
```